### PR TITLE
Modal anim

### PR DESCRIPTION
**Pull request by Cool04ek**

### CODE REVIEW

File: lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/ModalStack.java [hunk 1]
```diff
@@ -26,25 +27,33 @@
...
+        Modal modal = findModalByComponentId(componentId);
+        if (modal != null) {
+            modal.dismiss(promise);
+        } else {
+            Navigator.rejectPromise(promise);
+        }
+    }
+
+    @RestrictTo(RestrictTo.Scope.TESTS)
+    void dismissAll() {
```
Comment (guyca) [inline lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/ModalStack.java]: This is not the correct way to mock a function. Adding test specific logic is wrong.

File: lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/ModalStack.java [hunk 2]
```diff
@@ -26,25 +27,33 @@
...
-	void dismissAll(Promise promise) {
-		for (Modal modal : modals) {
-			modal.dismiss(size() == 1 ? promise : new NoOpPromise());
-		}
-		modals.clear();
-	}
+    }
+
+    void dismissModal(final String componentId, Promise promise) {
+        Modal modal = findModalByComponentId(componentId);
```
Comment (guyca) [inline lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/ModalStack.java]: Would be nice to add a `applyOnModal` method which will accept three parameters,
1. componentId
2. accept method
3. reject method

This way we avoid the messy null checks

File: lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/Modal.java [hunk 3]
```diff
@@ -19,24 +23,44 @@
...
+
     public Modal(final ViewController viewController, ModalListener modalListener) {
         this.viewController = viewController;
         dialog = new Dialog(viewController.getActivity(), R.style.Modal);
         this.modalListener = modalListener;
         dialog.setOnKeyListener(this);
         dialog.setOnDismissListener(this);
         dialog.setOnShowListener(this);
+        animator = new ModalAnimator(viewController.getActivity());
+        animator.setOptions(viewController.options.animationsOptions);
```
Comment (guyca) [inline lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/Modal.java]: Options should be passed in constructor as they are used to configure the animator.

File: lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/Modal.java [hunk 4]
```diff
@@ -19,24 +23,44 @@
...
         dialog.show();
+        animator.animateShow(contentView, new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationStart(Animator animation) {
+                dialog.setContentView(contentView);
+            }
+        });
+    }
+
+    @RestrictTo(RestrictTo.Scope.TESTS)
```
Comment (guyca) [inline lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/Modal.java]: This method needs to be remove. We can't add methods or any logic for tests.
The only acceptable use for `RestrictTo(RestrictTo.Scope.TESTS)` is for getters/setters. This way we avoid using tools like [Power Mock](https://github.com/powermock/powermock).

File: lib/android/app/src/main/res/values/styles.xml [hunk 5]
```diff
@@ -3,10 +3,11 @@
 
     <style name="Modal" parent="Theme.AppCompat.Light.NoActionBar">
         <item name="android:windowAnimationStyle">@style/modalAnimations</item>
+        <item name="android:windowBackground">@android:color/transparent</item>
     </style>
 
     <style name="modalAnimations">
-        <item name="android:windowEnterAnimation">@anim/slide_up</item>
-        <item name="android:windowExitAnimation">@anim/slide_down</item>
+        <item name="android:windowEnterAnimation">@null</item>
```
Comment (guyca) [inline lib/android/app/src/main/res/values/styles.xml]: Nice 👍

File: lib/android/app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -3,10 +3,11 @@
 
     <style name="Modal" parent="Theme.AppCompat.Light.NoActionBar">
         <item name="android:windowAnimationStyle">@style/modalAnimations</item>
+        <item name="android:windowBackground">@android:color/transparent</item>
```
Comment (guyca) [inline lib/android/app/src/main/res/values/styles.xml]: Nice 👍

File: lib/android/app/src/test/java/com/reactnativenavigation/viewcontrollers/ModalStackTest.java [hunk 7]
```diff
@@ -103,7 +104,7 @@ public void isEmpty() throws Exception {
     public void onDismiss() throws Exception {
         uut.showModal(viewController, new MockPromise());
         uut.showModal(new SimpleViewController(newActivity(), "otherComponent", new Options()), new MockPromise());
-        uut.dismissAll(new MockPromise());
+        uut.dismissAll();
```
Comment (guyca) [inline lib/android/app/src/test/java/com/reactnativenavigation/viewcontrollers/ModalStackTest.java]: What's the reason behind this change?

File: playground/src/app.js [hunk 8]
```diff
@@ -24,7 +24,7 @@ function start() {
   registerScreens();
   Navigation.events().onAppLaunched(() => {
     Navigation.setDefaultOptions({
-      _animations: {
+      animations: {
```
Comment (guyca) [inline playground/src/app.js]: Lets avoid using custom animations for now, or at least make the animation look decent. :)

Comment (guyca) [inline playground/src/app.js]: Lets avoid using custom animations in the playground app for now.
If you really want to use them, let's make the animation look decent :)

File: lib/android/app/src/test/java/com/reactnativenavigation/viewcontrollers/ModalStackTest.java (hunk 7 above)
Comment (Cool04ek) [reply lib/android/app/src/test/java/com/reactnativenavigation/viewcontrollers/ModalStackTest.java]: Tests do not wait for animation to finish. And we dismiss dialog in the onAnimationEnd callback. So I'm not sure I know how to fix this test.

File: lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/ModalStack.java (hunk 1 above)
Comment (Cool04ek) [reply lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/ModalStack.java]: +
